src/DynamoCoreWpf/Interfaces/IWatchHandler.cs

```diff
@@ -50,36 +54,44 @@ public DefaultWatchHandler(IPreferences preferences)
             this.preferences = preferences;
         }
 
-        private WatchViewModel ProcessThing(object value, ProtoCore.RuntimeCore runtimeCore, string tag, bool showRawData, WatchHandlerCallback callback)
+        private WatchViewModel ProcessThing(object value, ObservableCollection<PortModel> outports, ProtoCore.RuntimeCore runtimeCore, string tag, bool showRawData, WatchHandlerCallback callback)
         {
             if (value is DesignScript.Builtin.Dictionary)
             {
                 var dict = value as DesignScript.Builtin.Dictionary;
+
                 var keys = dict.Keys;
                 var values = dict.Values;
+                if (outports.Count > 1)
+                {
+                    keys = outports.Select(p => p.Name);
+                    values = keys.Select(k => dict.ValueAtKey(k));
+                }
```

Comment:
A corner case is a multi-return node that returns a `Dictionary` on a port. We should ensure that the keys are matching.

Reply:
You're right I'm not quite sure how to maintain the ordering for a nested Dictionary like so:
![image](https://user-images.githubusercontent.com/5710686/35405921-82a7cb08-01d5-11e8-8b00-1a862f3bfdfa.png)

@kronz @Racel, any comments?

Reply:
@pboyer @aparajit-pratap - Yes, it is strange that they do not match. I agree we should try to ensure that the order is matching. But, I do not think this should hold up a release .If this is too difficult to figure out, we can go back and fix in 2.1. It will be an edge case after all...We dont have any OOTB nodes like this.